starlette/testclient.py

```diff
@@ -382,7 +382,7 @@ def __init__(
         self.app = asgi_app
         self.base_url = base_url
 
-    def request(
+    def request(  # type: ignore
```

Comment:
Mypy sees the `url` parameter of the superclass method `Session.request` as a `Union[str, bytes]`, so it is complaining about the signature here. But the `urljoin` call below will fail if it `self.base_url` is a `str` and `url` is a `bytes`, so we probably shouldn't annotate it as `Union[str, bytes]`.

For what it's worth, mypy does let you use error codes now, so we could now make the `# type: ignore` more specific if desired. I have no preference here.